### PR TITLE
Remove dependency on deprecated github.com/pkg/errors

### PR DESCRIPTION
**Pull request by zchee**
#### Summary

Remove dependency on deprecated github.com/pkg/errors.

#### Ticket Link

Update: https://github.com/sigstore/cosign/issues/1880

#### Release Note

???

### CODE REVIEW

**Comment by zchee** [comment]
cc: @imjasonh I started cleanup pkg/errors dependency, but still draft.

File: cmd/cosign/cli/commands.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func New() *cobra.Command {
 				var err error
 				out, err = os.Create(ro.OutputFile)
 				if err != nil {
-					return errors.Wrapf(err, "Error creating output file %s", ro.OutputFile)
+					return fmt.Errorf("error creating output file %s: %w", ro.OutputFile, err)
```
Comment (zchee) [inline cmd/cosign/cli/commands.go:66]: also to lowercase for Go idiom, np?

Comment (imjasonh) [reply cmd/cosign/cli/commands.go:66]: Yeah I think we should lowercase the error messages, but otherwise don't make any changes to them.

**Comment by imjasonh** [comment]
FYI, also chasing down some usages in dependencies:
- https://github.com/awslabs/amazon-ecr-credential-helper/pull/321
- https://github.com/open-policy-agent/opa/pull/4696

**Comment by zchee** [comment]
@imjasonh Great, let's remove deprecated packages from the Go community world :P

**Review by imjasonh — APPROVED** [review]

**Comment by zchee** [comment]
@imjasonh 
I wrote this change by hand this time and it was very annoying lol
I'll write an analyzer that rewrites source code using "x/tools/go/analysis" and send a pull request to fulcio and rekor, with it. It would be useful in other projects as well.

Well, PTAL.

File: cmd/cosign/cli/commands.go (hunk 1 above)
Comment (zchee) [reply cmd/cosign/cli/commands.go:66]: act.

**Comment by imjasonh** [comment]
https://github.com/sigstore/cosign/runs/6458613293?check_suite_focus=true#step:12:286 looks like a bad `%w` usage.

I've noticed when doing this in other repos that sometimes you'll have a `errors.Wrapf("hello", someString)` (without a `%s`), which nothing complains about. But then when you change it to `fmt.Errorf` things notice the missing placeholder and complain, or fail, or have bad output. It's good to find them, but it's one more annoying thing to have to look for.

**Comment by zchee** [comment]
@imjasonh Ah, mistake, will fix asap.

> Oh please oh please oh please. If you do this I will be forever in your debt. 🙏

lol, will do.

File: pkg/cosign/kubernetes/webhook/validator.go [hunk 2]
```diff
@@ -433,7 +433,7 @@ func ValidatePolicySignaturesForAuthority(ctx context.Context, ref name.Referenc
 		// https://github.com/sigstore/cosign/issues/1652
 		sps, err := valid(ctx, ref, rekorClient, authority.Key.PublicKeys, remoteOpts...)
 		if err != nil {
-			return nil, errors.Wrap(err, fmt.Sprintf("failed to validate public keys with authority %s for %s", name, ref.Name()))
+			return nil, fmt.Errorf(fmt.Sprintf("failed to validate public keys with authority %s for %s: %w", name, ref.Name()), err)
```
Comment (zchee) [inline pkg/cosign/kubernetes/webhook/validator.go:436]: will fix too.

**Comment by zchee** [comment]
@imjasonh 
> sigstore/cosign/runs/6458613293?check_suite_focus=true#step:12:286 looks like a bad %w usage.
>
> I've noticed when doing this in other repos that sometimes you'll have a errors.Wrapf("hello", someString) (without a %s), which nothing complains about. But then when you change it to fmt.Errorf things notice the missing placeholder and complain, or fail, or have bad output. It's good to find them, but it's one more annoying thing to have to look for.

I found the root cause that `GetEnvTargetRepository` did not handle error. Fixed on [`80ab7ef` (#1887)](https://github.com/sigstore/cosign/pull/1887/commits/80ab7efc8751bd1b8c04b14f1e16cbcf4eb1ca43). PTALA.

**Review by dlorenc — APPROVED** [review]

**Review by cpanato — APPROVED** [review]

**Comment by zchee** [comment]
@imjasonh @dlorenc @cpanato thanks! I'll fix rekor and fulcio too.

**Pull request merged**